resources/qml/UserSettingsPage.qml

```diff
@@ -0,0 +1,218 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+// SPDX-FileCopyrightText: 2022 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import "ui"
+import Qt.labs.platform 1.1 as Platform
+import QtQuick 2.15
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.2
+import QtQuick.Window 2.13
+import im.nheko 1.0
+
+Rectangle {
+    id: userSettingsDialog
+
+    property bool collapsed: width < 800
+    color: Nheko.colors.window
+
+    Shortcut {
+        sequence: StandardKey.Cancel
+        onActivated: userSettingsDialog.close()
+    }
+
+    ScrollView {
+        id: scroll
+
+        palette: Nheko.colors
+        ScrollBar.horizontal.visible: false
+        anchors.fill: parent
+        anchors.margins: Nheko.paddingLarge
+
+        contentWidth: availableWidth
+
+        Timer {
+            id: deadTimer
+            interval: 500
+        }
+
+        Connections {
+            target: scroll.contentItem
+            function onContentYChanged() { deadTimer.restart(); }
+        }
+
+
+        GridLayout {
+            id: grid
+
+            columns: userSettingsDialog.collapsed ? 1 : 2
```

Comment:
I think this causes the actual control for changing a setting to be on a different line than the setting description itself when in collapsed mode, which isn't consistent with how the dialog currently resizes.  It also looks a little awkward, in my opinion, having all of that blank space to the right of the description, a new line, and then a ton of blank space to the left of the control on the next line.

Reply:
My scale factor is 1, so I don't think it should break anything.

Reply:
For me it only collapses around here:

![grafik](https://user-images.githubusercontent.com/16312324/148847840-dd9d7c17-8164-4aca-adbb-97831061c0c6.png)

Reply:
Which is a bit early, but any later and the content would overflow, because some lines would not fit in the grid, especially in German.